src/robotFactory.js

```diff
@@ -1,10 +1,80 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, position = {}, chipVersion) {
+    const { x = 0, y = 0 } = position;
 
-class FlyingRobot {}
+    this.name = name;
+    this.weight = weight;
 
-class DeliveryDrone {}
+    this.coords = {
+      x: x,
+      y: y,
+    };
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    return (this.coords.y += step);
```

Comment:
```suggestion
    this.coords.y += step
```
looks like no need to return anything